pkg/typesystem/typesystem.go

```diff
@@ -190,7 +192,9 @@ func New(model *openfgav1.AuthorizationModel) *TypeSystem {
 
 	uncompiledConditions := make(map[string]*condition.EvaluableCondition, len(model.GetConditions()))
 	for name, cond := range model.GetConditions() {
-		uncompiledConditions[name] = condition.NewUncompiled(cond)
+		uncompiledConditions[name] = condition.NewUncompiled(cond).
+			WithMaxEvaluationCost(defaultMaxEvaluationCost).
```

Comment:
Not sure the best composition for this. I'm open to suggestions.

This should be configurable, but in order to make it configurable in a dynamic way with our current `EvaluableCondition` design we have to provide it before we compile the condition. If we rework our `Evaluate` method we could construct the CEL program in the Evaluate method and provide this more dynamic at evaluation time instead of compile time.

It didn't feel too out of place to put this here because we're controlling model behavior, and so having it in the Typesystem package didn't seem too unreasonable, but it doesn't' feel "perfect" to me.

Reply:
i see what you mean, but i don't dislike it tbh. NewUncompiled is just a constructor and in that sense this kinda fits in there nicely.